keycloak-user-api/src/main/java/io/enmasse/user/keycloak/KeycloakUserApi.java

```diff
@@ -47,16 +48,24 @@
     private final Duration apiTimeout;
     private final Map<String, Keycloak> keycloakMap = new HashMap<>();
 
-    public KeycloakUserApi(KeycloakFactory keycloakFactory, Clock clock) {
-        this(keycloakFactory, clock, Duration.ZERO);
-    }
-
     public KeycloakUserApi(KeycloakFactory keycloakFactory, Clock clock, Duration apiTimeout) {
         this.keycloakFactory = keycloakFactory;
         this.clock = clock;
         this.apiTimeout = apiTimeout;
     }
 
+    public synchronized void pruneAuthenticationServices(List<AuthenticationService> items) {
```

Comment:
I found the API less than obvious.  On my first reading of the use, I was expecting the items list to be list that is being pruned rather than those that are to be retained.

Reply:
Fixed

Reply:
That does the job.